i4mi/midata.js
# Update angular dependencies to match those of ionic 5

**Summary**
---
Development with ionic 5 is not compatible with the current version of midata.js since they both run on different major releases of angular. This should be considered as a bug and can be fixed with the proposal of this pull request.

**WARNING:** This pull request has not been tested for backwards compatibility. Ionic projects being developed at the institute must be checked if they use an ionic version lower than 5.0.x and/or an ```@angular/cli``` lower than 7.2.x

**Steps to reproduce**
---
1) Install the latest version of ionic, which currently is 5.0.0:
```
npm install -g ionic@latest
```

2) Create a new ionic project:
```
ionic start myApp blank
```

3) Add the current version of midata.js to your new project and try to start it without deploying (this is necessary to make sure that ionic uses the `@angular-devkit` dependencies)
```
ionic cordova run android --device --no-native-run
```

**Working example**
---
The latest commit on the branch **'midata'** from [this project](https://gitlab.ti.bfh.ch/cleanapp/mobile) provides a working example of the pull request. The bug can be reproduced using the following command:
```
npm run debug:android 
```

**What is the current bug behavior?**
---
Following error output will be visible in the console:
```node
[ng] ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
[ng] node_modules/midata/node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
```

The old version of `@angular/devkit` tries to declare a variable which has already been declared by ionic. This causes any developer action (such as `ionic start`) to fail.
Deploying the project as an apk is not affected.

The package of midata.js should use the same version of angular as the current version of ionic. If the problem still persists, midata.js should not use angular at all to avoid unwanted overlaps.
